casper/src/test/resources/RevAddressTest.rho

```diff
@@ -62,5 +64,20 @@ in {
       revAddress!("fromPublicKey", "F0F0".hexToBytes(), *retCh) |
       rhoSpec!("assert", (Nil, "== <-", *retCh), "correct RevAddress", *ackCh)
     }
+  } |
+
+  contract test_fromUnforgeable(rhoSpec, _, ackCh) = {
+    new unf1, unf2, r1, r2 in {
+      revAddress!("fromUnforgeable", *unf1, *r1) |
+      revAddress!("fromUnforgeable", *unf2, *r2) |
+      rhoSpec!("assert", (*r1, "!= <-", *r2), "expecting RevAddresses to be unequal", *ackCh)
+    }
+  } |
+
+  contract test_fromUnforgeable_invalid(rhoSpec, _, ackCh) = {
+    new retCh in {
+      revAddress!("fromUnforgeable", 42, *retCh) |
```

Comment:
We should add a similar test for revAddress!fromDeployerId that passes a non-deployerId (e.g. gprivate)

Reply:
I thought about cleaning up all system processes in a separate PR but I am kind of blocked by not being sure on how to return an either from withing a system process. Any ideas?

Reply:
`An Either is just a coproduct of two cases encoded as a tuple, what's the problem?` :stuck_out_tongue: 

I too would have to look up how to create `(false, "error msg")` or `(true, result)` in AST :)

Reply:
Anyway, I think such a cleanup can cause an avalanche of changes. Let's then ticket it and for now leave as is.

Reply:
Ouch! Was totally complicating it. Thanks...

Reply:
Btw there is no `RevAddress!fromDeployerId` :)

Reply:
Right! There should be! :joy:

Reply:
I now see that we have deployerId -> pubKey and pubKey -> revAddress, so having deployerId -> revAddress would rather be a convenience than a necessity:)